src/io/sendmsg.rs

```diff
@@ -0,0 +1,103 @@
+use crate::buf::BoundedBuf;
+use crate::io::SharedFd;
+use crate::runtime::driver::op::{Completable, CqeResult, Op, SingleCQE};
+use crate::runtime::CONTEXT;
+use socket2::SockAddr;
+use std::io;
+use std::io::IoSlice;
+use std::net::SocketAddr;
+
+pub(crate) struct SendMsg<T, U> {
+    _fd: SharedFd,
+    _io_bufs: Vec<T>,
+    _io_slices: Vec<IoSlice<'static>>,
+    _socket_addr: Option<Box<SockAddr>>,
+    msg_control: Option<U>,
+    msghdr: libc::msghdr,
+}
+
+impl<T: BoundedBuf, U: BoundedBuf> Op<SendMsg<T, U>, SingleCQE> {
```

Comment:
Super small nit. Does it still compile if you leave out the SingleCQE here? I think that would make it consistent with other files in this src.

Just to keep complexity down, we often want to just follow other examples when there is more than one right way to do things.